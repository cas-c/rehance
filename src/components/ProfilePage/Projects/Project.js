import React from 'react';
import PropTypes from 'prop-types';

const Project = ({ data }) => (
    <a href={data.url} target='_blank' title={data.name} >
        <div className='card'>
            <header className='card-header'>
                <p className='card-header-title'>
                    { data.name }
                </p>
            </header>
            <div className='card-image project-cover'>
                <figure className='image is-4by3'>
                    {
                        data.covers && <img className='project-cover-img' src={data.covers.original} alt={data.name} />
                    }
                </figure>
            </div>
        </div>
    </a>
);

Project.propTypes = {
    data: PropTypes.shape({
        url: PropTypes.string,
        name: PropTypes.string,
        covers: PropTypes.shape({
            original: PropTypes.string
        })
    })
}

Project.defaultProps = {
    data: {}
}

export default Project;
