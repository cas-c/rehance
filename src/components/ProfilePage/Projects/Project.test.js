import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Project from './Project';

const mockProject = {
    url: 'Mock URL',
    name: 'Test Project Name',
    covers: {
        original: 'Original cover URL'
    }
}

describe('Project', () => {
    it('snapshot: should render correctly', () => {
        const output = shallow(<Project data={mockProject} />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('snapshot: should not error out without input', () => {
        const output = shallow(<Project />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
    it('unit: should only display cover art if cover info exists', () => {
        const mounted = mount(<Project data={mockProject} />);
        expect(mounted.containsMatchingElement(<img className='project-cover-img' src='Original cover URL' alt='Test Project Name' />));
        mounted.unmount();
    })
});
