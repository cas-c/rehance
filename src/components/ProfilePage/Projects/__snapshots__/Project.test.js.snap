// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project snapshot: should not error out without input 1`] = `
<a
  target="_blank"
>
  <div
    className="card"
  >
    <header
      className="card-header"
    >
      <p
        className="card-header-title"
      />
    </header>
    <div
      className="card-image project-cover"
    >
      <figure
        className="image is-4by3"
      />
    </div>
  </div>
</a>
`;

exports[`Project snapshot: should render correctly 1`] = `
<a
  href="Mock URL"
  target="_blank"
  title="Test Project Name"
>
  <div
    className="card"
  >
    <header
      className="card-header"
    >
      <p
        className="card-header-title"
      >
        Test Project Name
      </p>
    </header>
    <div
      className="card-image project-cover"
    >
      <figure
        className="image is-4by3"
      >
        <img
          alt="Test Project Name"
          className="project-cover-img"
          src="Original cover URL"
        />
      </figure>
    </div>
  </div>
</a>
`;
